src/pybind/mgr/telemetry/module.py

```diff
@@ -230,6 +298,29 @@ def gather_mon_metadata(self,
 
         return metadata
 
+    def gather_mds_metadata(self) -> Dict[str, Dict[str, int]]:
+        metadata: Dict[str, Dict[str, int]] = dict()
+
+        res = self.get('mds_metadata')  # metadata of *all* mds daemons
+        if res is None or not res:
+            self.log.debug('Could not get metadata for mds daemons')
+            return metadata
+
+        keys = list()
+        keys += self.metadata_keys
+
+        for key in keys:
+            metadata[key] = defaultdict(int)
+
+        for mds in res.values():
+            for k, v in mds.items():
+                if k not in keys:
+                    continue
+
+                metadata[k][v] += 1
```

Comment:
Aha, so you wanted to count things, uh? So why not using a [Counter](https://docs.python.org/3/library/collections.html#collections.Counter)?

```py
metadata[key] = Counter()

metadata[key][v] += 1
```